webui/react/src/pages/TrialDetails/Profiles/ProfilesEnabled.tsx

```diff
@@ -23,29 +29,42 @@ const ProfilesEnabled: React.FC<Props> = ({ trial }: Props) => {
     filters.gpuUuid,
   );
 
-  return (
-    <>
+  // memoize if trial has profiling data
+  useEffect(() => {
+    if (hasProfilingData || systemMetrics.isEmpty) return;
+    setHasProfilingData(true);
+  }, [ hasProfilingData, setHasProfilingData, systemMetrics.isEmpty ]);
 
-      {systemMetrics.isEmpty && (
-        <Alert
-          message="No data available."
-          type="warning"
-        />
-      )}
+  if (!hasProfilingData) {
```

Comment:
**suggestion:** I noticed this `No data available.` alert gets shown as an initial state (even while loading). Does it make sense to change it to something like:
```
  if (!hasProfilingData && !systemMetrics.isLoading && !timingMetrics.isLoading) {
```
So that it shows a spinner while it's attempting to load.

Reply:
Not really at the moment, because `isLoading` is set to false only after receiving events from the streaming endpoint.

We could potentially consider adding a timeout say after 5/10s to force `isLoading` to be false independently from receiving events or not?